docs/standard/collections/commonly-used-collection-types.md

```diff
@@ -15,15 +15,68 @@ ms.assetid: f5d4c6a4-0d7b-4944-a9fb-3b12d9ebfd55
+
+In collections based on <xref:System.Collections.IList>, or directly on <xref:System.Collections.ICollection>, every element contains only a value. Examples include:
+
+- <xref:System.Array>
+- <xref:System.Collections.ArrayList>
+- <xref:System.Collections.Generic.List%601> 
+- <xref:System.Collections.Queue>
+- <xref:System.Collections.Concurrent.ConcurrentQueue%601>
+- <xref:System.Collections.Stack>
+- <xref:System.Collections.Concurrent.ConcurrentStack%601>
+- <xref:System.Collections.Generic.LinkedList%601>
+
+In collections based on the <xref:System.Collections.IDictionary> interface, every element contains both a key and a value. Examples include:
+
+- <xref:System.Collections.Hashtable> 
+- <xref:System.Collections.SortedList>  
+- <xref:System.Collections.Generic.SortedList%602> 
+- <xref:System.Collections.Generic.Dictionary%602>  
+- <xref:System.Collections.Concurrent.ConcurrentDictionary%602>  
+
+The <xref:System.Collections.ObjectModel.KeyedCollection%602> class is unique because it is a list of values with keys embedded within the values. As a result, it behaves both like a list and like a dictionary.  
+
+When you need efficient multi-threaded collection access, use the generic collections in the <xref:System.Collections.Concurrent> namespace.
+
+## Creating generic collections
+Generic collections are the best solution to strong typing. A `List<xref:System.Int32>`, for instance, will present a compile-time error if you add any other type to it.
+ 
+If your language does not support generics, the <xref:System.Collections> namespace includes abstract base classes that you can extend to create strongly-typed collections: 
+
+- <xref:System.Collections.CollectionBase>
+- <xref:System.Collections.ReadOnlyCollectionBase>
+- <xref:System.Collections.DictionaryBase>
+
+## How collections vary
   
- Generic collections are the best solution to strong typing. However, if your language does not support generics, the <xref:System.Collections> namespace includes base collections, such as <xref:System.Collections.CollectionBase>, <xref:System.Collections.ReadOnlyCollectionBase>, and <xref:System.Collections.DictionaryBase>, which are abstract base classes that can be extended to create collection classes that are strongly typed. When efficient multi-threaded collection access is required, use the generic collections in the <xref:System.Collections.Concurrent> namespace.  
+Collections vary depending on how they store elements, how they perform searches, how they compare elements, and how they are sorted. 
+
+All collections use zero-based indexes except <xref:System.Array>, which allows arrays that are not zero-based.  
+
+### Sorting
+You cannot sort the <xref:System.Collections.Hashtable> or the <xref:System.Collections.Generic.Dictionary%602> classes. The <xref:System.Collections.SortedList> class and the generic <xref:System.Collections.Generic.SortedList%602> provide sorted versions of these collections.
```

Comment:
I don't think that's true. You can call OrderBy to sort dictionaries and hash tables. I would just keep the original version of this sentence.